src/main/java/org/fenixedu/academic/domain/student/Registration.java

```diff
@@ -3430,6 +3430,14 @@ public boolean hasMissingPersonalInformationForAcademicService(ExecutionYear exe
         return true;
     }
 
+    @Atomic
+    public void createReingression(ExecutionYear executionYear, LocalDate reingressionDate) {
+        RegistrationDataByExecutionYear dataByYear =
+                RegistrationDataByExecutionYear.getOrCreateRegistrationDataByYear(this, executionYear);
+        dataByYear.createReingression(reingressionDate);
+    }
+
+    @Deprecated
     public boolean isReingression(final ExecutionYear executionYear) {
```

Comment:
Given that this PR is targeting a major version, you may safely delete this method (it seems to me that it is unused anyway).